src/themes/default/components/core/ProductGallery.vue

```diff
@@ -63,34 +38,44 @@
 
 <script>
 import { ProductGallery } from '@vue-storefront/core/modules/catalog/components/ProductGallery.ts'
-import ProductGalleryZoom from './ProductGalleryZoom'
+import ProductGalleryOverlay from './ProductGalleryOverlay'
 import NoSSR from 'vue-no-ssr'
 import VueOfflineMixin from 'vue-offline/mixin'
+const ProductGalleryCarousel = () => import(/* webpackChunkName: "vsf-product-gallery-carousel" */ './ProductGalleryCarousel.vue')
 
 export default {
   components: {
+    ProductGalleryCarousel,
     'no-ssr': NoSSR,
-    ProductGalleryZoom
+    ProductGalleryOverlay
   },
   mixins: [ProductGallery, VueOfflineMixin],
   watch: {
     '$route': 'validateRoute'
   },
   data () {
     return {
-      loaded: true,
-      carouselTransitionSpeed: 0
+      isZoomOpen: false,
+      showProductGalleryCarousel: false,
+      currentSlide: 0
     }
   },
+  mounted () {
+    this.showProductGalleryCarousel = true
+  },
   methods: {
+    openOverlay (currentSlide) {
+      this.currentSlide = currentSlide
+      this.toggleZoom()
+    },
     validateRoute () {
       this.$forceUpdate()
     }
   }
 }
 </script>
 
-<style lang="scss" scoped>
+<style lang="scss">
```

Comment:
why you had to remove scoped style?

Reply:
yeah, that was not very good. Because i separated the component in childcomponents quiet a bit, the styling was not correct anymore. I fixed this in a new commit on this branche.